src/policy/fees.h

```diff
@@ -196,20 +197,18 @@ class CBlockPolicyEstimator
     /** Remove a transaction from the mempool tracking stats*/
     void removeTx(std::map<uint256, TxStatsInfo>::iterator pos, bool inBlock);
 
+    /** Helper method to quickly remove transactions that left mempool without going in a block */
+    void removeTxNotInBlock(const uint256& hash);
+
 public:
     /** Create new BlockPolicyEstimator and initialize stats tracking classes with default values */
     CBlockPolicyEstimator();
     ~CBlockPolicyEstimator();
 
-    /** Process all the transactions that have been included in a block */
-    void processBlock(unsigned int nBlockHeight,
-                      const std::vector<CTransactionRef>& txn_removed_in_block);
-
-    /** Process a transaction accepted to the mempool*/
-    void processTransaction(const CTxMemPoolEntry& entry, bool validFeeEstimate);
-
-    /** Remove a transaction from the mempool tracking stats*/
-    bool removeTx(uint256 hash);
+    // MempoolInterface functions:
+    void TransactionAddedToMempool(const NewMempoolTransactionInfo& info, const std::vector<CTransactionRef>& txn_replaced) override;
```

Comment:
commit: Process fee estimates in the background via MempoolInterface

Make the callbacks protected?

Reply:
Fixed.